app/javascript/bundles/CourseNavBar/components/CourseNavBar.jsx

```diff
@@ -0,0 +1,49 @@
+import React, {Component, Fragment} from 'react';
+import * as PropTypes from 'prop-types';
+import {Row, Col, MenuItem, Tab, Nav, NavItem, NavDropdown, Button} from "react-bootstrap";
+
+class CourseNavBar extends Component {
+    currentActiveTab = () => {
+        // Captures path after the course index.
+        // E.g. for path '/courses/1/project_teams/new', this captures 'project_teams'.
+        const pathMatchRegex = new RegExp('\/courses\/[0-9]*/([^/]*)', 'i');
+        const coursePathMatches = pathMatchRegex.exec(this.props.current_path);
+        if (coursePathMatches == null || coursePathMatches.length <= 1) {
+            return 'roster_students';
+        }
+        return coursePathMatches[1];
+    }
```

Comment:
I wonder if it would be better to factor this out into a method that takes a parameter so that it:
* can be unit tested
* so that the unit tests more clearly document what's going on here?

I envision something like this:

function currentActiveTab(path) {
   // same code, except instead of the quasi-global vairable this.props.current_path,
  // use the parameter path 

}
```
Then, line 18 would be:

const activeTabKey = currentActiveTab(this.props.current_path);